presto-main/src/main/java/com/facebook/presto/sql/gen/JoinCompiler.java

```diff
@@ -103,6 +121,26 @@ public LookupSourceFactory compileLookupSourceFactory(List<? extends Type> types
         }
     }
 
+    public JoinFilterFunctionVerifierFactory compileJoinFilterFunctionVerifierFactory(Optional<JoinFilterFunction> joinFilterFunction)
```

Comment:
I'm curious why we have `JoinFilterFunction`abstraction at all if we really want a `JoinFilterFunctionVerifier`.  The we could simply generate a `JoinFilterFunctionVerifier`, which is basically the existing code with on extra `leftPages` field, and this method:
```java
    public boolean applyFilterFunction(int leftBlockIndex, int leftPosition, int rightPosition, Block[] allRightBlocks)
    {
        return filterFunction.filter(leftPosition, leftPages.get(leftBlockIndex), rightPosition, allRightBlocks);
    }